lib/membrane_rtc_engine/endpoints/hls_endpoint.ex

```diff
@@ -63,77 +95,187 @@ if Enum.all?(
                   spec: pid(),
                   description: "Pid of parent Engine"
                 ],
-                output_directory: [
-                  spec: Path.t(),
-                  description: "Path to directory under which HLS output will be saved",
-                  default: "hls_output"
-                ],
                 owner: [
                   spec: pid(),
                   description: """
                   Pid of parent all notifications will be send to.
-
                   These notifications are:
-                    * `{:playlist_playable, content_type, stream_id, origin}`
+                  * `{:playlist_playable, content_type}`
+                  * `{:cleanup, clean_function}`
                   """
                 ],
-                hls_mode: [
-                  spec: :separate_av | :muxed_av,
-                  default: :separate_av,
+                output_directory: [
+                  spec: Path.t(),
+                  description: "Path to directory under which HLS output will be saved",
+                  default: "hls_output"
+                ],
+                mixer_config: [
+                  spec: MixerConfig.t() | nil,
+                  default: nil,
                   description: """
-                  Defines output mode for `Membrane.HTTPAdaptiveStream.SinkBin`.
-
-                  - `:separate_av` - audio and video tracks will be separated
-                  - `:muxed_av` - audio will be attached to every video track
+                  Audio and video mixer configuration. If you don't want to use compositor pass nil.
                   """
                 ],
-                target_window_duration: [
-                  type: :time,
-                  spec: Membrane.Time.t() | :infinity,
-                  default: Membrane.Time.seconds(20),
+                sink_bin_config: [
```

Comment:
We shouldn't name this sink_bin_config. Users will not have any idea what sink bin is